File: 1614. Maximum Nesting Depth of the Parentheses/maxDepth.js
/**
 * @param {string} s
 * @return {number}
 */
const maxDepth = (s) => {
  let res = 0;
  let cur = 0;

  for (let i = 0; i < s.length; i++) {
    switch (s[i]) {
      case '(':
        cur++;
        break;
      case ')':
        cur--
        break;
    }

    res = Math.max(res, cur);
  }

  return res;
};


const s = "(1+(2*3)+((8)/4))+1";
console.log('maxDepth: ', maxDepth(s));

File: 1614. Maximum Nesting Depth of the Parentheses/maxDepth.py
class Solution:
  def maxDepth(self, s: str) -> int:
    res = 0
    cur = 0
    
    for char in s:
      if char == '(':
        cur += 1
      elif char == ')':
        cur -= 1
      res = max(res, cur)  
      
    return res
    

s = "(1+(2*3)+((8)/4))+1"
solution = Solution()
print('maxDepth: ', solution.maxDepth(s))
